src/constructs/ServerSideWebsite.ts

```diff
@@ -0,0 +1,436 @@
+    }
+
+    async getDistributionId(): Promise<string | undefined> {
+        return this.provider.getStackOutput(this.distributionIdOutput);
+    }
+
+    getMainCustomDomain(): string | undefined {
+        if (this.configuration.domain === undefined) {
+            return undefined;
+        }
+
+        // In case of multiple domains, we take the first one
+        return typeof this.configuration.domain === "string" ? this.configuration.domain : this.configuration.domain[0];
+    }
+
+    private headersToForward(): OriginRequestHeaderBehavior {
+        let additionalHeadersToForward = this.configuration.forwardedHeaders ?? [];
+        if (additionalHeadersToForward.includes("Host")) {
+            throw new ServerlessError(
+                `Invalid value in 'constructs.${this.id}.forwardedHeaders': the 'Host' header cannot be forwarded (this is an API Gateway limitation). Use the 'X-Forwarded-Host' header in your code instead (it contains the value of the original 'Host' header).`,
+                "LIFT_INVALID_CONSTRUCT_CONFIGURATION"
+            );
+        }
+        if (additionalHeadersToForward.includes("Authorization")) {
+            // `Authorization` cannot be forwarded via this setting (we automatically forward it anyway so we remove it from the list)
+            additionalHeadersToForward = additionalHeadersToForward.filter(
+                (header: string) => header !== "Authorization"
+            );
+        }
+
+        /**
+         * We forward everything except:
+         * - `Host` because it messes up API Gateway (that uses the Host to identify which API Gateway to invoke)
+         * - `Authorization` because it must be configured on the cache policy
+         *   (see https://aws.amazon.com/premiumsupport/knowledge-center/cloudfront-authorization-header/?nc1=h_ls)
+         */
+        return OriginRequestHeaderBehavior.allowList(
+            "Accept",
+            "Accept-Language",
+            "Origin",
+            "Referer",
+            // This header is set by our CloudFront Function
+            "X-Forwarded-Host",
+            // We merge the user-provided list in the hardcoded list
+            ...additionalHeadersToForward
+        );
+    }
+
+    private createCacheBehaviors(s3Origin: S3Origin): Record<string, BehaviorOptions> {
+        const behaviors: Record<string, BehaviorOptions> = {};
+        for (const pattern of Object.keys(this.getAssetPatterns())) {
+            behaviors[pattern] = {
+                // Origins are where CloudFront fetches content
+                origin: s3Origin,
+                allowedMethods: AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
+                // Use the "Managed-CachingOptimized" policy
+                // See https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-managed-cache-policies.html#managed-cache-policies-list
+                cachePolicy: CachePolicy.CACHING_OPTIMIZED,
```

Comment:
See https://github.com/getlift/lift/discussions/43#discussioncomment-979124

Reply:
I did additional testing here and it seems we are good: https://github.com/getlift/lift/discussions/43#discussioncomment-1249228